samples/net/google_iot_mqtt/src/main.c

```diff
@@ -0,0 +1,196 @@
+/* Full-stack IoT client example. */
+
+/*
+ * Copyright (c) 2018 Linaro Ltd
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(net_google_iot_mqtt, LOG_LEVEL_INF);
+
+#include <zephyr.h>
+
+#include "dhcp.h"
+#include "protocol.h"
+
+#include <net/sntp.h>
+#include <net/net_config.h>
+#include <net/net_event.h>
+
+/* This comes from newlib. */
+#include <time.h>
+#include <inttypes.h>
+
+#ifdef CONFIG_STDOUT_CONSOLE
+# include <stdio.h>
+# define PRINT printf
+#else
+# define PRINT printk
+#endif
+
+struct k_sem sem;
+
+static s64_t time_base;
+
+void resp_callback(struct sntp_ctx *ctx,
+		   int status,
+		   u64_t epoch_time)
+{
+	s64_t stamp;
+	char time_str[sizeof("1970-01-01T00:00:00")];
+
+	if (status == 0) {
+		stamp = k_uptime_get();
+		time_base = epoch_time * MSEC_PER_SEC - stamp;
+
+		/* Convert time to make sure. */
+		time_t now = epoch_time;
+		struct tm now_tm;
+
+		gmtime_r(&now, &now_tm);
+		strftime(time_str, sizeof(time_str), "%FT%T", &now_tm);
+		LOG_INF("  Acquired time: %s", log_strdup(time_str));
+
+	} else {
+		LOG_ERR("  Failed to acquire SNTP, code %d\n", status);
+	}
+
+	k_sem_give(&sem);
+}
+
+/* Zephyr implementation of POSIX `time`.  Has to be called k_time
+ * because time is already taken by newlib.  The clock will be set by
+ * the SNTP client when it receives the time.  We make no attempt to
+ * adjust it smoothly, and it should not be used for measuring
+ * intervals.  Use `k_uptime_get()` directly for that.   Also the
+ * time_t defined by newlib is a signed 32-bit value, and will
+ * overflow in 2037.
+ */
+time_t k_time(time_t *ptr)
```

Comment:
You are stealing the kernel namespace here. Why not call it my_k_time() or similar instead?

Reply:
changed and moved to protocol.c